tests/Integration/Behaviour/Features/Context/Domain/CarrierRangesFeatureContext.php

```diff
@@ -0,0 +1,108 @@
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierRangesCollection;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+class CarrierRangesFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @Then I set ranges for carrier :reference called :newReference with specified properties for all shops:
+     */
+    public function setCarrierRangesAllShops(string $reference, string $newReference, TableNode $node): void
+    {
+        $this->setCarrierRanges($reference, $newReference, ShopConstraint::allShops(), $node);
+    }
+
+    /**
+     * @Then I set ranges for carrier :reference called :newReference with specified properties for shop :shopReference:
+     */
+    public function setCarrierRangesShop(string $reference, string $newReference, string $shopReference, TableNode $node): void
+    {
+        $this->setCarrierRanges($reference, $newReference, $this->getShopConstraint($shopReference), $node);
+    }
+
+    /**
+     * @Then carrier :reference should have the following ranges for all shops:
+     */
+    public function getCarrierRangesAllShops(string $reference, TableNode $node): void
+    {
+        $this->getCarrierRanges($reference, ShopConstraint::allShops(), $node);
+    }
+
+    /**
+     * @Then carrier :reference should have the following ranges for shop :shop:
+     */
+    public function getCarrierRangesShop(string $reference, string $shopReference, TableNode $node): void
+    {
+        $this->getCarrierRanges($reference, $this->getShopConstraint($shopReference), $node);
+    }
+
+    private function setCarrierRanges(string $reference, string $newReference, ShopConstraint $shopConstraint, TableNode $node): void
+    {
+        try {
+            $carrierId = $this->referenceToId($reference);
+
+            $command = new SetCarrierRangesCommand($carrierId, $node->getColumnsHash(), $shopConstraint);
+
+            $carrierId = $this->getCommandBus()->handle($command);
+            $this->getSharedStorage()->set($newReference, $carrierId->getValue());
+        } catch (CarrierException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    private function getCarrierRanges(string $reference, ShopConstraint $shopConstraint, TableNode $node): void
+    {
+        try {
+            $carrierId = $this->referenceToId($reference);
+
+            $command = new GetCarrierRanges($carrierId, $shopConstraint);
+
+            $rangesDatabase = $this->getCommandBus()->handle($command);
+            $rangesExpected = new CarrierRangesCollection($node->getColumnsHash());
+
+            Assert::assertEquals($rangesExpected, $rangesDatabase);
+        } catch (CarrierException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    private function getShopConstraint(string $shopReference): ShopConstraint
+    {
+        return ShopConstraint::shop((int) $this->referencesToIds($shopReference));
```

Comment:
```suggestion
        return ShopConstraint::shop((int) $this->referenceToId($shopReference));
```
Be careful `referencesToIds` is used for lists of shops `shop1, shop3` and returns an array, here you only want ONE id